syntax = "proto3";

package hnstream;

service Hnstreamer {
  rpc GetNextComments(GetNextRequest) returns (GetNextCommentsResponse) {}
  rpc GetNextStories(GetNextRequest) returns (GetNextStoriesResponse) {}
}

message GetNextRequest {
  int64 unix_time = 1;
  int64 limit = 2;
}

message Story {
  int64 id = 1;
  int64 score = 2;
  int64 time = 3;
  string by = 4;
  string title = 5;
  string url = 6;
  string text = 7;
  string author = 8;
}

message Comment {
  int64 id = 1;
  int64 time = 2;
  int64 parent = 3;
  int64 ranking = 4;
  string by = 5;
  string author = 6;
  string text = 7;
}

message GetNextStoriesResponse {
  repeated Story stories = 1; 
}

message GetNextCommentsResponse {
  repeated Comment comments = 1; 
}
